core/constants/env/env.go

```diff
@@ -70,6 +70,7 @@ const VSecMSidecarSuccessThreshold VarName = "VSECM_SIDECAR_SUCCESS_THRESHOLD"
 const VSecMSpiffeIdPrefixSafe VarName = "VSECM_SPIFFEID_PREFIX_SAFE"
 const VSecMSpiffeIdPrefixSentinel VarName = "VSECM_SPIFFEID_PREFIX_SENTINEL"
 const VSecMSpiffeIdPrefixWorkload VarName = "VSECM_SPIFFEID_PREFIX_WORKLOAD"
+const VSecMWorkloadNameRegExp VarName = "VSECM_WORKLOAD_NAME_REGEXP"
```

Comment:
This is new. It gives us the flexibility to capture any part of the SPIFFE ID as the workload’s id.

One restriction is; all regexp shall start with `^spifee://$trustDomain/` where $trustDomain is whichever trust domain has been configured.

It is also recommended for the regExps to end with a `$` and make the matches as specific as possible whenever possible.